pynitrokey/helpers.py

```diff
@@ -34,6 +36,32 @@
 STDOUT_PRINT = True
 
 
+def filter_sensitive_parameters(parameters):
+    """Helper function to remove sensitive parameters
+
+    `parameters`: `list of str`
+
+    Returns:
+        `list of str`: `parameters` without sensitive values listed
+                       in `pynitrokey.confconsts.CLI_LOG_BLACKLIST`
+    """
+    parameters = list(
+        chain.from_iterable(
+            [parameter.split("=", maxsplit=1) for parameter in parameters]
+        )
+    )
+
+    for sensitive_parameter, parameter_length in CLI_LOG_BLACKLIST:
+        if sensitive_parameter in parameters:
+            sensitive_index = parameters.index(sensitive_parameter)
+            parameters = (
+                parameters[: sensitive_index + 1]
+                + ["[redacted]"]
+                + parameters[sensitive_index + 1 + parameter_length :]
+            )
```

Comment:
What do you think about this version?  This would keep the number of arguments for a command.

```python
for i in range(sensitive_index + 1, sensitive_index + 1 + parameter_length):
    parameters[i] = "[redacted]"
```

Reply:
I now use this version as I also think keeping the number of arguments improves the log readability even more.